sycl/doc/extensions/proposed/sycl_ext_oneapi_composite_device.asciidoc

```diff
@@ -0,0 +1,290 @@
+
+The APIs defined in this extension are only useful when using the Level Zero
+backend, and they are only useful when the Level Zero environment variable
+`ZE_FLAT_DEVICE_HIERARCHY=COMBINED` is set.  The APIs may be called even when
+using other backends, but they will return an empty list of composite devices.
+
+
+== Overview
+
+Some Intel GPU architectures are structured with multiple tiles on a single
+card.  Currently, this applies only to the Data Center GPU Max series (aka
+PVC).  By default, SYCL exposes each of these tiles as a separate root device,
+so each tile corresponds to a separate `device` object that is returned from
+`device::get_devices`.  Applications generally perform better when each tile
+is treated as a separate device, but some advanced use cases can benefit by
+treating all tiles on a card as a single "composite" device.  This extension
+provides APIs that enable these advanced use cases.
+
+A composite device has the same semantics as any other SYCL device, though the
+performance characteristics might be different.  The application may submit a
+kernel to a composite device, and the implementation automatically schedules
+work-items to each of the underlying tiles.  Memory allocated on the composite
+device is accessible to any of these work-items, regardless of which underlying
+tile it runs on.
+
+Applications that use this extension can access the same hardware through
+different `device` objects.  For example, it is possible to get one `device`
+object that represents a card and another `device` object that represents a
+tile from that card.  As noted previously, a device representing a card is
+known as a "composite" device.  A device representing a tile is known as a
+"component" of that composite device.  Since the tile-level devices are
+returned from `device::get_devices`, they are also considered "root" devices,
+which is a term defined in the core SYCL specification.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_COMPOSITE_DEVICE` to one of the values defined in the
+table below.  Applications can test for the existence of this macro to
+determine if the implementation supports this feature, or applications can test
+the macro's value to determine which of the extension's features the
+implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== Enumerating composite devices
+
+This extension adds two new functions for enumerating the available composite
+devices.  One is a free function and the other is a new member function of the
+`platform` class:
+
+```
+namespace sycl {
+namespace ext::oneapi::experimental {
+
+std::vector<device> get_composite_devices();
```

Comment:
@gmlueck : from the definition above:

>A composite device has the same semantics as any other SYCL device, though the
performance characteristics might be different.  The application may submit a
kernel to a composite device, and the implementation automatically schedules
work-items to each of the underlying tiles.

then `get_composite_devices()` would return the equivalent devices returned by `zeDeviceGet` when ZE_FLAT_DEVICE_HIERARCHY is set to composite. When it is set to FLAT, then `get_composite_devices()`  and `zeDeviceGet` will return 0, as with FLAT the root devices returned are tiles, so the statement above "the implementation automatically schedules work-items to each of the underlying tiles." doesn't hold.

Is that correct?